code/modules/mob/living/carbon/diona_base.dm

```diff
@@ -50,6 +50,11 @@ var/list/diona_banned_languages = list(
 	DS.last_location = loc
 	if(light_amount)
 		adjustNutritionLoss(-light_amount) // regenerate our nutrition from light.
+	move_delay_mod = max(initial(move_delay_mod) - light_amount * 1.5, 0)
+	species.sprint_cost_factor = max(min(initial(species.sprint_cost_factor), initial(species.sprint_cost_factor) - light_amount * DIONA_LIGHT_COEFICIENT), 0)
```

Comment:
There is only one instance of `species` per every species datum. You do _not_ want to modify the variables of this instance.